src/data_structures/fmindex.rs

```diff
@@ -168,18 +183,46 @@ impl FMDIndex {
     ///   less memory usage, but worse performance)
     pub fn new(bwt: BWT, k: usize) -> Self {
         let mut alphabet = dna::n_alphabet();
-        alphabet.insert(b'$');
-        assert!(
-            alphabet.is_word(&bwt),
-            "Expecting BWT over the DNA alphabet (including N) with the sentinel $."
-        );
+        alphabet.insert(SENT_CHAR);
+        assert!(alphabet.is_word(&bwt),
+                "Expecting BWT over the DNA alphabet (including N) with the sentinel $.");
 
         FMDIndex {
             fmindex: FMIndex::new(bwt, k, &alphabet),
-            revcomp: dna::RevComp::new()
+            revcomp: dna::RevComp::new(),
         }
     }
 
+    /// Find longest non-encompassing prefix match
+    pub fn longest_prefix_match(&self, pattern: &[u8], min_overlap: usize) -> Option<BiInterval> {
+        let exact_match_interval = self.fmindex.backward_search(pattern.iter());
```

Comment:
Okay. I've tested it, and it does what I expected (it's fine that it considers reverse complements-- it should, since reads can occur from either direction during sequencing). I've committed a new version with the name change, and also changed it to return the length of the overlap. I can take a look later and try and match it more precisely to the Sanger pseudocode, but it probably won't be until later this week or early next week. If you have time yourself, feel free to make whatever changes you see fit.